tests/integration.rs

```diff
@@ -362,6 +362,25 @@ fn test_version_sort_overwrite_by_sizesort() {
         .stdout(predicate::str::is_match("11\n2\n$").unwrap());
 }
 
+#[test]
+#[cfg(target_os = "linux")]
+fn test_bad_utf_8_extension() {
+    Command::new("sh")
+        .arg("-c")
+        .arg("mkdir -p /tmp/bad.extension/")
+        .output()
+        .expect("failed to create dir");
+    Command::new("python3")
+        .arg("-c")
+        .arg(r#"import pathlib; pathlib.Path('/tmp/bad.extension/bad.extension\udca7\udcfd').touch()"#)
+        .output()
+        .expect("failed to create file");
```

Comment:
Hey man it may be best to avoid having python as a dependency, this can be done in sh with
```shell
$ sh -c "touch $(printf 'bad.extension\xdca7\xdcfd')"
```
```suggestion
    Command::new("sh")
    	.arg("-c")
    	.arg("touch $(printf '/tmp/bad.extension/bad.extension\xdca7\xdcfd')")
    	.output()
    	.expect("failed to create file");
```

Reply:
and this would also fix the CI since it's not set up to install python

Reply:
I don't want the python dep either, but I can not find a way to create the invalid utf-8 name even use your code.🤣

I have tried the solution like this, The output of the `sh -c touch` is iliteral `\xdca7`, not an utf-8 code.

invalid utf-8 code will be treated as error if using double quote.

```
---- test_bad_utf_8_extension stdout ----
thread 'test_bad_utf_8_extension' panicked at 'Unexpected stdout, failed var.is_match(bad.extension��
$)
└── var as str: bad.extension\xdca7\xdcfd


command=`"/root/lsd/target/debug/lsd" "/tmp/.tmpRDHlAb"`
code=0
stdout=```bad.extension\xdca7\xdcfd
```
stderr=``````
',
```

Reply:
I was still trying to find a way to create the invalid name by `shell`, but no luck, python is my last choice...